### PR TITLE
Create login via Slack

### PR DESCRIPTION
**Pull request by tuxagon**
Testing this is a little tricky because you will need to supply the `SLACK_OAUTH_TOKEN` and `SLACK_SIGNING_SECRET` first to ensure you can talk to Slack and receive requests from Slack. You'll also need to enable a way for Slack to send a request to your running instance. If you're developing locally, `ngrok` is helpful.

Run `ngrok http 3000` and start your rails server with `NGROK_DOMAIN=<Add ngrok domain here> bin/rails s`.

At that point, simply run `/doubleup login` which should trigger an auth link to be sent in Slack DMs. That link is then the last step to authenticating into Double Up.

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -10,26 +10,34 @@
...
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
   end
 
   create_table "pending_notifications", force: :cascade do |t|
     t.string "strategy"
     t.date "last_attempted_on"
     t.bigint "historical_match_id"
-    t.datetime "created_at", precision: 6, null: false
-    t.datetime "updated_at", precision: 6, null: false
```
Comment (jasoncaryallen) [inline db/schema.rb:31]: Curious/non blocking Q: I feel like this is an OK change, and encountered similar on my WIP branch (#58) - is there a reason this was set previously or carryover of local a `db:migrate`?

**Review by jasoncaryallen — APPROVED** [review]
Looks good from what I see here and what was discussed while pairing yesterday! One quick Q, non-blocking - otherwise 🚢 🇮🇹

File: db/schema.rb (hunk 1 above)
Comment (tuxagon) [reply db/schema.rb:31]: I don't really remember sadly. It's likely I overlooked adding the change awhile back, but not really sure.

**Comment by tuxagon** [comment]
Only thing this requires is a new oauth scope before merging for deployment.

**Pull request merged**